x-pack/plugins/cases/public/components/all_cases/all_cases_generic.tsx

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { EuiProgress } from '@elastic/eui';
+import { EuiTableSelectionType } from '@elastic/eui/src/components/basic_table/table_types';
+import { isEmpty, memoize } from 'lodash/fp';
+import styled, { css } from 'styled-components';
+import classnames from 'classnames';
+
+import {
+  Case,
+  CaseStatuses,
+  CaseType,
+  CommentRequestAlertType,
+  CommentType,
+  FilterOptions,
+  SortFieldCase,
+  SubCase,
+} from '../../../common';
+import { SELECTABLE_MESSAGE_COLLECTIONS } from '../../common/translations';
+import { useGetActionLicense } from '../../containers/use_get_action_license';
+import { useGetCases } from '../../containers/use_get_cases';
+import { usePostComment } from '../../containers/use_post_comment';
+import { CaseCallOut } from '../callout';
+import { CaseDetailsHrefSchema, CasesNavigation } from '../links';
+import { Panel } from '../panel';
+import { getActionLicenseError } from '../use_push_to_service/helpers';
+import { ERROR_PUSH_SERVICE_CALLOUT_TITLE } from '../use_push_to_service/translations';
+import { useCasesColumns } from './columns';
+import { getExpandedRowMap } from './expanded_row';
+import { CasesTableHeader } from './header';
+import { CasesTableFilters } from './table_filters';
+import { EuiBasicTableOnChange } from './types';
+
+import { CasesTable } from './table';
+const ProgressLoader = styled(EuiProgress)`
+  ${({ $isShow }: { $isShow: boolean }) =>
+    $isShow
+      ? css`
+          top: 2px;
+          border-radius: ${({ theme }) => theme.eui.euiBorderRadius};
+          z-index: ${({ theme }) => theme.eui.euiZHeader};
+        `
+      : `
+      display: none;
+    `}
+`;
+
+const getSortField = (field: string): SortFieldCase =>
+  field === SortFieldCase.closedAt ? SortFieldCase.closedAt : SortFieldCase.createdAt;
+
+interface AllCasesGenericProps {
+  alertData?: Omit<CommentRequestAlertType, 'type'>;
+  caseDetailsNavigation?: CasesNavigation<CaseDetailsHrefSchema, 'configurable'>; // if not passed, case name is not displayed as a link (Formerly dependant on isSelector)
+  configureCasesNavigation?: CasesNavigation; // if not passed, header with nav is not displayed (Formerly dependant on isSelector)
+  createCaseNavigation: CasesNavigation;
+  disabledStatuses?: CaseStatuses[];
+  isSelector?: boolean;
```

Comment:
I've been back and forth on this one and it's probably a nit, but wondering if we can change the name of this or add view or something like: `isSelectorView` or `modalSelector` or something. The only reason being I forsee confusion with redux `selectors` as cases grows